javascript/readme.md

```diff
@@ -1,3 +1,386 @@
+- The word function is always followed with one space.
+- No space should separate a unary operator and its operand except when the operator is a word such as typeof.
+- All binary operators should be separated from their operands by a space on each side except `.` period and `(` left parenthesis and `[` left bracket.
+- Every `,` comma should be followed by a space or a line break.
+- Each `;` semicolon at the end of a statement should be followed with a line break.
+- Each `;` semicolon in the control part of a for statement should be followed with a space.
+
+Every statement should begin aligned with the current indentation. The outermost level is at the left margin. The indentation increases by 4 spaces when the last token on the previous line is { left brace, [ left bracket, ( left paren. The matching closing token will be the first token on a line, restoring the previous indentation.
+
+The ternary operator can be visually confusing, so ? question mark always begins a line and increase the indentation by 4 spaces, and : colon always begins a line, aligned with the ? question mark.
+
+```javascript
+var integer = function (
+    value,
+    default_value
+) {
+    value = resolve(value);
+    return (typeof value === "number")
+        ? Math.floor(value)
+        : (typeof value === "string")
+            ? value.charCodeAt(0)
+            : default_value;
+};
+```
+
+If `.` period is the first character on a line, the indentation is increased by 4 spaces.
+
+Avoid excessively long lines. When a statement will not fit nicely on a single line, it may be necessary to break it. It is best to break after a `{` left brace, `[` left bracket, `(` left paren, `,` comma, or before a `.` period, `?` question mark, or `:` colon. If such a break is not feasible, then break after an operator and continue on the next line with 8 spaces added to the current indentation. Those 8 spaces do not change the current indentation.
+
+Clauses (`case`, `catch`, `default`, `else`, `finally`) are not statements and so should not be indented like statements.
+
+Tabs and spaces should not be mixed. We should pick just one in order to avoid the problems that come from having both. Personal preference is an extremely unreliable criteria. Neither the tab nor the space offers a powerful advantage over the other. Fifty years ago, tab had the advantage of consuming less memory, but Moore's Law has eliminated that advantage. Space has one clear advantage over tab: there is still no reliable standard for how many spaces a tab represents, but it is universally accepted that a space occupies a space. So always use spaces. You can edit with tabs if you must, but make sure it is spaces again before you commit. Maybe someday we will finally get a universal standard for tabs, but until that day comes, the better choice is spaces.
```

Comment:
Not sure we need this in as much detail (Moore's Law etc.). Really we should be using the settings in the editorconfig only.